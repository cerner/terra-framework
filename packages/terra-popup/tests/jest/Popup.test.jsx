import React from 'react';
import PopupDefaultExample from './PopupDefaultExample';

// Snapshot Tests
it('should shallow a default component', () => {
  const popup = shallow(<PopupDefaultExample />);
  expect(popup).toMatchSnapshot();
});

it('should shallow render a component with push attachment behavior', () => {
  const popup = shallow(<PopupDefaultExample attachmentBehavior="push" />);
  expect(popup).toMatchSnapshot();
});

it('should renders with custom arrow class name', () => {
  const popup = shallow(<PopupDefaultExample classNameArrow="terra-test-class-arrow" />);
  expect(popup).toMatchSnapshot();
});

it('should renders with custom content class name', () => {
  const popup = shallow(<PopupDefaultExample classNameContent="terra-test-class-content" />);
  expect(popup).toMatchSnapshot();
});

it('should renders with custom overlay class name', () => {
  const popup = shallow(<PopupDefaultExample classNameOverlay="terra-test-class-overlay" />);
  expect(popup).toMatchSnapshot();
});
