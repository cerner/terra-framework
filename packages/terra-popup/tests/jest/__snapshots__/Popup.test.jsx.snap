// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should shallow a default component 1`] = `
<div>
  <Popup
    attachmentBehavior="auto"
    boundingRef={null}
    classNameArrow={null}
    classNameContent={null}
    classNameOverlay={null}
    contentAttachment="top center"
    contentHeight="80"
    contentWidth="240"
    isArrowDisplayed={false}
    isContentFocusDisabled={false}
    isHeaderDisabled={false}
    isOpen={true}
    onRequestClose={[Function]}
    targetRef={[Function]}
  >
    <p>
      this is popup content
    </p>
  </Popup>
  <div
    style={
      Object {
        "height": "20px",
        "width": "20px",
      }
    }
  />
</div>
`;

exports[`should shallow render a component with push attachment behavior 1`] = `
<div>
  <Popup
    attachmentBehavior="push"
    boundingRef={null}
    classNameArrow={null}
    classNameContent={null}
    classNameOverlay={null}
    contentAttachment="top center"
    contentHeight="80"
    contentWidth="240"
    isArrowDisplayed={false}
    isContentFocusDisabled={false}
    isHeaderDisabled={false}
    isOpen={true}
    onRequestClose={[Function]}
    targetRef={[Function]}
  >
    <p>
      this is popup content
    </p>
  </Popup>
  <div
    style={
      Object {
        "height": "20px",
        "width": "20px",
      }
    }
  />
</div>
`;
